Improve routing API

fpseverino opened the pull request:
- Make `PassesService` and `OrdersService` conform to `RouteCollection`
    - Services can now be registered as controllers on any `RoutesBuilder`
- Use the request's `Logger` inside routes (instead of one provided by the user)
- Remove push routes and `pushRoutesMiddleware`
   - Those were somewhat useful in early versions of the package when methods like `sendPushNotifications` were not public

fpseverino committed:
Improve routing API

ptoffy commented on Sources/VaporWalletOrders/OrdersServiceCustom+RouteCollection.swift:
```diff
@@ -0,0 +1,182 @@
+            throw Abort(.badRequest)
+        }
+        guard
+            let order = try await O.query(on: req.db)
```
These (`O`, `D` etc.) don't seem like great names for models

fpseverino replied:
An alternative could be `OrderType`, `DeviceType`, etc.

0xTim replied:
Yes this is a terrible name for anyone trying to read. Make it explicit

ptoffy reviewed (commented):
Where'd the tests go?

fpseverino commented:
> Where'd the tests go?

I removed only the tests related to the two "push routes", which I would like to remove

The first one when called sent a push notification to all devices registered to a specific pass, the other one returned as a response the push tokens of all devices registered to a pass

Their usefulness is questionable in my opinion, and their initialization required (rightly) a middleware, that with this refactoring of the routes into a `RouteCollection` should have been saved inside `PassesService`/`OrdersService`

P.S. Of course we already provide the methods that these routes used internally and they are easily reimplementable by the user if they need them

ptoffy commented:
I'm not against removing useless stuff but we should be testing public stuff so if you remove tests I'd remove the tested stuff too. When reviewing it seemed like you only removed the tests and not the routes

0xTim reviewed (changes requested):
Definitely need better variable names. I'm assuming all the new push stuff is tested?

One thing I want to check now that we're relying on route registration is that it works with nested stuff unlike the Imperial issues?

fpseverino committed:
Rename generic types

fpseverino commented:
> I'm not against removing useless things but we should be testing what's public so if you remove tests I'd remove the tested routes too. When reviewing it seemed like you only removed the tests and not the routes

Yes, I also removed push routes, you can see that service initializers no longer accept the [`pushRoutesMiddleware` argument](https://github.com/vapor-community/wallet/pull/20/files#diff-10658eca8334b2ca2944110471887e407b46aefa5777e7fd64d630deb8225cc3L22).
I also moved all routes to a new file, so maybe it's hard to notice that the push ones are no longer there

fpseverino commented:
> I'm assuming all the new push stuff is tested?

The push notifications public methods haven't changed (they were already tested), I just removed the so called "push routes" which were a duplicate way to send notifications (not required by Apple Wallet API specification)

fpseverino commented:
> One thing I want to check now that we're relying on route registration is that it works with nested stuff unlike the Imperial issues?

The difference between this package and Imperial is that Imperial automatically returns the URL to the API based on the registered route, whereas here the route registration and the URL encoding are two completely separate things.

So it works perfectly in nested routes, but you have to manually build the URL when encoding the pass.

For example, if you register the routes like this:
```swift
try app.grouped("api", "passes").register(collection: passesService)
```
you have to set the `webServiceURL` of the pass like this:
```swift
let webServiceURL = "\(Environment.get("WEBSITE_URL")!)/api/passes/"
```

If we find a solution to this common problem we could think of automatically injecting the path

fpseverino committed:
Update README for v0.7.0

fpseverino committed:
Update SPI links
